Description: Cut month name so it fits into the output.
Author: Michael Meskes <meskes@debian.org>

--- bsdmainutils/usr.bin/ncal/ncal.c	2009-11-02 16:10:43.000000000 +0100
+++ bsdmainutils/usr.bin/ncal/ncal.c	2009-11-02 16:08:58.000000000 +0100
@@ -546,7 +552,11 @@
 
 	mw = jd_flag ? MONTH_WIDTH_B_J : MONTH_WIDTH_B;
 
-	swprintf(s, MAX_WIDTH, L"%ls %d", month.name, y);
+	swprintf(s, MAX_WIDTH, L"%d", y);
+	i = wcslen(s);
+	swprintf(s, MAX_WIDTH, L"%ls", month.name);
+	s[mw - i -1] = '\0';
+	swprintf(s + wcslen(s), MAX_WIDTH, L" %d", y);
 	wprintf(L"%ls\n", wcenter(t, s, mw));
 
 	if (jd_flag)
